core/src/main/kotlin/resolvers/LocationProviderFactory.kt

```diff
@@ -0,0 +1,13 @@
+package org.jetbrains.dokka.resolvers
+
+import org.jetbrains.dokka.pages.ModulePageNode
+import org.jetbrains.dokka.pages.PageNode
+import org.jetbrains.dokka.plugability.DokkaContext
+
+interface LocationProviderFactory {
+    fun getLocationProvider(pageNode: PageNode, context: DokkaContext): LocationProvider
```

Comment:
I am aware that it will look awkward here but to be consistent we shouldn't pass `DokkaContext` to `getLocationProvider`. If creator of custom location provider wants to use context api in their factory, they can always use lazyly evaluated extension.

Reply:
Done